[Magic WAN] sophos guide (#8380)
cloudflare/cloudflare-docs: content/magic-wan/tutorials/sophos-firewall.md

---
title: Sophos Firewall

> pcx_content_type: tutorial

Reviewer: ```suggestion
pcx_content_type: integration-guide
```

Reviewer: Not too sure about this one, but based on this https://wiki.cfdata.org/x/dYY0FQ, it seemed it could make sense

Author: Good point! This has been a back and forth since before we created this content type :)

---

# Sophos Firewall

> This tutorial will show you how to use Magic WAN with the following versions of the Sophos Firewall:

Reviewer: ```suggestion
This tutorial shows you how to use Magic WAN with the following versions of the Sophos Firewall:
```

- **Sophos form factor tested:**
    - Sophos Firewall XGS and XG series hardware
    - Sophos Firewall virtual appliance on VMware 

- **Sophos software versions tested:**
    - SFOS  Version 19.0 MR2-Build 472
    - SFOS  Version 19.5.1 MR1-Build 278

> You can connect through Generic Routing Encapsulation (GRE) or IPsec tunnels to Magic WAN.

Reviewer: Would it make sense to add some links to `Generic Routing Encapsulation (GRE)` and `IPsec tunnels`?

## IPsec connection

The following instructions show how to setup an IPSec connection on your Sophos Firewall device.

### 1. Add an IPsec Profile

1. Go to **System** > **Profiles**.
2. In **IPsec profiles**, select **Add**.

> 3. In the **General settings** group, make sure you have the following settings:

Reviewer: Is the `Re-key connection` option and other fields that appear in the screenshot selected by default or is it up to the user? Maybe, it could be useful to clarify that since we are only guiding the user about a few settings they should make sure to have but we do not say anything about the rest of it.

    - **Name**: Give your site-to-site VPN a descriptive name.
    - **Connection type**: _Tunnel interface_
    - **Gateway type**: _Initiate the connection_
4. In the **Encryption** group, make sure you have the following settings:
    - **Authentication type**: **Preshared key**
5. In **Gateway settings**, make sure you have the following settings:
    - **Gateway address**: Enter your Cloudflare Anycast IP address provided by Cloudflare.

![Configure an IPsec tunnel.](/magic-wan/static/sophos-firewall/2-ipsec-tunnel.png)

After setting up your IPsec tunnel, it will show up on the IPsec connections list with an **Active** status.

![The IPsec tunnel should show up on the IPsec connections list.](/magic-wan/static/sophos-firewall/2b-ipsec-tunnel.png)

### 3. Assign the XFRM interface address

You must use an interface address from the `/31` subnet required to [configure tunnel endpoints](/magic-wan/get-started/configure-tunnels/) on Magic WAN.

1. Go to **Configure** > **Network**.
2. In **Interfaces**, select the corresponding interface to the IPsec tunnel you created in [step 2](#2-create-ipsec-connection-tunnel).

> 3. Edit the interface to assign an address from the `/31` subnet required to [configure tunnel endpoints](/magic-wan/get-started/configure-tunnels/). 

Reviewer: I was a bit confused about the connection between this step and what the screenshot shows. Is it an example of how the final result should look like after the user has already edited it?

Below are instruction on how to achieve this on SFOS version 19 and SFOS version 19.5:

#### SFOS 19.0 MR2-Build 472 or 19.5 MR1-Build278 or later versions: 

1. Sign in to the CLI.
2. Enter **4** to choose **Device console**, and enter the following command:

    ```bash
    set vpn ipsec-performance-setting anti-replay window-size 0
    ```

    ![Access the CLI to disable anti-replay](/magic-wan/static/sophos-firewall/5-sfos-19.png)

#### Older SFOS versions

Contact Sophos support.

## GRE connection

### 1. Configure a GRE tunnel between SFOS and Cloudflare

Start by configuring a GRE tunnel between SFOS and the Cloudflare Anycast IP address.

1. Sign in to the CLI. 
2. Enter **4** to choose **Device console**, and enter the following command:

    ```bash
    system gre tunnel add name <NAME_OF_YOUR_GRE_TUNNEL> local-gw <WAN_PORT> remote-gw <REMOTE_GATEWAY_IP_ADDRESS> local-ip <LOCAL_IP_ADDRESS> remote-ip <REMOTE_IP_ADDRESS>
    ```

    ![Access the CLI to configure a GRE tunnel](/magic-wan/static/sophos-firewall/1-gre-connection.png)

>     For more details, refer to [KB-000035813](https://support.sophos.com/support/s/article/KB-000035813?language=en_US).

Reviewer: ```suggestion
    For more details, refer to the [Sophos Firewall knowledge base](https://support.sophos.com/support/s/article/KB-000035813?language=en_US).
```

1. Go to **Protect** > **Rules and policies**.
2. In **Firewall rules**, select **IPv4** > **Add firewall rule**.

![Create a firewall rule with the criteria and security policies from your company](/magic-wan/static/sophos-firewall/4-firewall-rule.png)

## Traffic redirection mechanism on Sophos Firewall

To redirect traffic, you can add a static or an SD-WAN route.

### IPsec

#### Static route 

Go to **Configure** > **Routing** > **Static routes** to add an XFRM interface-based route. The interface will be automatically created when you set up a tunnel interface based on IPsec (such as the Cloudflare_MWAN example from above).

![Go to static routes to add an XFRM interface-based route](/magic-wan/static/sophos-firewall/static-route.png)

#### SD-WAN route 

1. Go to **Configure** > **Routing** > **Gateways** to create a custom gateway on the XFRM interface. The interface will be automatically created when you set up a tunnel interface based on IPsec (such as the Cloudflare_MWAN example from above).

![Go to Gateways to add an XFRM interface-based route](/magic-wan/static/sophos-firewall/1-sd-wan-gateway.png)

> 2. In **Configure** > **Routing** > **SD-WAN routes**, select **Add** to add the desired networks and services in the route to redirect traffic to Cloudflare. Enter a descriptive name for your connection, and pay attention to the **Incoming interface**, **Source networks** and **Primary gateway options**.

Reviewer: Maybe specify a bit more what the user should `pay attention to` about these fields? Is it in the sense of having it consistent with some other setting?

13:09:55.500480 xfrm1, OUT: IP 172.64.240.252 > 172.70.51.31: ICMP echo reply, id 33504, seq 0, length 64 

13:09:55.504669 xfrm1, IN: IP 172.71.29.66 > 172.64.240.252: ICMP echo request, id 60828, seq 0, length 64 
13:09:55.504695 xfrm1, OUT: IP 172.64.240.252 > 172.71.29.66: ICMP echo reply, id 60828, seq 0, length 64
```

## Verify tunnel status on Sophos Firewall dashboard

### IPsec

When the tunnel is working, its **Status** will be green.

![If the tunnel is working, it will show up with a green status.](/magic-wan/static/sophos-firewall/2b-ipsec-tunnel.png)

The corresponding XFRM interface will also show a **Connected** status.

![The XFRM interface will also show a connected status.](/magic-wan/static/sophos-firewall/1-sd-wan-gateway.png)

### GRE

Access the CLI and type `system gre tunnel show` to check the status of a GRE tunnel. When the tunnel is working, its Status will show up as **Enabled**.

![The GRE tunnel will show a status of Enabled when working.](/magic-wan/static/sophos-firewall/gre-status-enabled.png)

![The GRE tunnel will show a status of Enabled when working.](/magic-wan/static/sophos-firewall/gre-status-enabled-b.png)

> ## Troubleshooting

Reviewer: Don't know if that is the case but if some of the suggested steps below have their own page within the docs, I would consider linking them.
